include/device.h

```diff
@@ -250,38 +240,28 @@ struct device_pm {
 };
 
 /**
- * @brief Static device information (In ROM) Per driver instance
+ * @brief Runtime device structure (in memory) per driver instance
  *
  * @param name name of the device
  * @param init init function for the driver
  * @param config_info address of driver instance config information
- */
-struct device_config {
-	const char *name;
-	int (*init)(struct device *device);
-#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
-	int (*device_pm_control)(struct device *device, u32_t command,
-				 void *context, device_pm_cb cb, void *arg);
-	struct device_pm *pm;
-#endif
-	const void *config_info;
-};
-
-/**
- * @brief Runtime device structure (In memory) Per driver instance
  * @param device_config Build time config information
  * @param driver_api pointer to structure containing the API functions for
- * the device type. This pointer is filled in by the driver at init time.
+ * the device type.
  * @param driver_data driver instance data. For driver use only
  */
 struct device {
-	const struct device_config *config;
+	const char *name;
+	const void *config_info;
 	const void *driver_api;
-	void *driver_data;
+	void * const driver_data;
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	int (*device_pm_control)(struct device *device, u32_t command,
```

Comment:
This should be renamed to just `pm_control`; that it appears within a `device` structure should be enough to disambiguate.  Also this makes it consistent with the existing `pm` field in the same structure.

Reply:
I am not going to touch PM stuff in this refactoring. So no name change here. But I agree this name is wrong, as well as first parameter (it should be struct device *dev, and not a name that is the same as the struct itself)